lib/ansible/modules/storage/purestorage/purefa_pgsnap.py

```diff
@@ -113,36 +127,47 @@ def get_pgroupvolume(module, array):
         return None
 
 
+def get_rpgsnapshot(module, array):
```

Comment:
Intead of making a new function just replace the existing get_pgroupvolume, because the existing one looks broken, as it doesn't work if volumes are removed from a PG, or if hosts are used as the PG source, the existing function doesn't work at all.  
```suggestion
def get_pgroupvolume(module, array):
try:

        snapvolname = module.params['name'] + "." + module.params['suffix'] + "." + module.params['restore']
        for snapvol in array.list_volumes(snap='true',names=snapvolname):
            if snapvol['name'] == snapvolname:
                return snapvolname
    except:
        return None
```

Reply:
Added checks to fail gracefully if selected PG selected doesn't exist or if the volume selected is not in the PG.